lib/buildkite/builder/extension_manager.rb

```diff
@@ -6,26 +6,34 @@ class ExtensionManager
 
       def initialize(context)
         @context = context
-        @extensions = []
+        @extensions = {}
```

Comment:
Do we have any cases where order of extensions matter? Ruby hashes maintain order, but curious

Reply:
yes, order of extensions matters